packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -73,7 +136,12 @@ export interface _TelemetryViewProps {
  * @internal
  */
 export function _TelemetryView(props: _TelemetryViewProps): React.ReactElement {
-	const { telemetryEvents } = props;
+	let { telemetryEvents } = props;
+	const [maxEventsToDisplay, setMaxEventsToDisplay] = React.useState<number>(100);
+
+	if (telemetryEvents.length > maxEventsToDisplay) {
+		telemetryEvents = telemetryEvents.slice(0, maxEventsToDisplay);
+	}
```

Comment:
One small change here: instead of overwriting the `telemetryEvents` variable (which should go back to being a `const`; we should never mutate thing received as `props` in a component), we should either a) create a new variable to hold the "filtered" list of telemetry events, or b) directly do `telemetryEvents.slice(0, maxEventsToDisplay).map((message, index) => { ... })` in line 171 below. After that and the few remaining outstanding comments I'd say we can merge this!

Reply:
I see. Added `filiteredTelemetryEvents` here.